python/ray/experimental/state/common.py

```diff
@@ -193,8 +196,36 @@ class State(StateSchema):
     # Returns {"column_a", "column_b"}
     s.columns()
     ```
+
+    In addition, the schema also provides a humanify abstract method to
+    convert the state object into something human readable, ready for printing.
+
+    Subclasses should override this method, providing logic to convert its own fields
+    to something human readable, packaged and returned in a dict.
+
+    Each field that wants to be humanified should include a 'format_fn' key in its
+    metadata dictionary.
     """
 
+    @classmethod
+    def humanify(cls, state: dict) -> dict:
+        """Convert the given state object into something human readable."""
+        if not cls.is_valid_state(state):
```

Comment:
Regarding your comment about not being able to use Humanify if a state dict is missing some columns:

I assumed humanify was an atomic operation. Since python doesn't enforce types, I was thinking that we only mutate the data (state_dict) if it was a valid data class (containing all necessary fields) passed in.

What do you think?

Reply:
We could just iterate through all the available columns and check if there's a humanify function, if so, mutate the data? 

I think requiring all columns is not ideal since people could only do it with `detail=True`

Reply:
Ahh I see. Ok, in that case, I can remove the check and the is_valid_state function.